### PR TITLE
ListSet intial implementation.

### PR DESCRIPTION
**Pull request by OStevan**
__Goals?__
Move to `ListSet` and enable one important proof.

__Description?__
- extracted a verifiable `ListSet` implementation.

__How does it improve on the current state of verified components?__
- one `faultyExpansion` is no longer `extern`

### CODE REVIEW

**Comment by OStevan** [comment]
Ignoring plug-in error.

**Comment by romac** [comment]
Nice!

File: light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/BlockchainSystem.scala [hunk 1]
```diff
@@ -2,14 +2,16 @@ package ch.epfl.ognjanovic.stevan.tendermint.verified.blockchain
...
+import ch.epfl.ognjanovic.stevan.tendermint.verified.types.Chain.Genesis
 import stainless.annotation._
 import stainless.lang._
+import stainless.proof.check
+import utils.ListSet
 
 object BlockchainSystem {
 
-  @ghost
+  @ghost @extern
```
Comment (romac) [inline light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/BlockchainSystem.scala:14]: Why is this extern now?

File: light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/FaultChecker.scala [hunk 2]
```diff
@@ -25,66 +24,55 @@ case class FaultChecker() {
...
+  def moreFaultyDoesNotHelp(faultChecker: FaultChecker, current: ListSet[Address], next: ListSet[Address]): Unit = {
     require(current.subsetOf(next))
   }.ensuring(_ =>
     forall((validator: ValidatorSet) => {
       faultyExpansion(faultChecker, next, current, validator)
       !faultChecker.isCorrect(validator, current) ==> !faultChecker.isCorrect(validator, next)
     }))
 
-  // TODO if isCorrect is inlined this lemma can be proved but calls to isCorrect at other locations fail
-  @extern
```
Comment (romac) [inline light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/FaultChecker.scala:37]: 😄

File: light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/BlockchainSystem.scala (hunk 1 above)
Comment (OStevan) [reply light-client-core/src/main/scala/ch/epfl/ognjanovic/stevan/tendermint/verified/blockchain/BlockchainSystem.scala:14]: It timeouts now. This is next on the cleanup list

**Pull request merged**